File: dsl-api/src/main/java/cbs/dsl/api/context/ParameterContext.java
package cbs.dsl.api.context;

import lombok.Builder;

import java.util.Map;

// TODO: remove
@Deprecated(forRemoval = true)
@Builder
public record ParameterContext<T>(
    String eventCode,
    Long workflowExecutionId,
    String performedBy,
    String dslVersion,
    Map<String, Object> eventParameters,
    T payload) {}

File: dsl-api/src/main/kotlin/cbs/dsl/api/context/ParameterContext.kt
package cbs.dsl.api.context

open class ParameterContext(
    eventCode: String,
    workflowExecutionId: Long,
    performedBy: String,
    dslVersion: String,
    val eventParameters: Map<String, Any>,
) : BaseContext(eventCode, workflowExecutionId, performedBy, dslVersion)
